Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -488,7 +489,14 @@ class TabTrayController: UIViewController, UITabBarDelegate, UICollectionViewDel
             } else {
                 cell.favicon.image = UIImage(named: "defaultFavicon")
             }
-            cell.background.image = tab.screenshot
+
+            if let img = tab.screenshot {
+                cell.background.image = img
+            } else if let thumbs = profile.thumbnails as? SDWebThumbnails {
+                let key = SDWebThumbnails.getKey(tab.url!)
+                cell.background.moz_getImageFromCache(key, cache: thumbs.cache, completed: { img, err, type, key in
+                })
+            }
```

Comment:
Heh, they're actually <1/10th the size, so that could be a big difference if we store a lot of thumbnails (taking the top 3:2 thumbnail from a 9:16 screen is 3/8, then half the width and height would be (3/8)/4, or 3/32).